cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -582,12 +619,46 @@ private String asLabel(String val) {
    * should always be ok.
    */
   private String cleanse(String val, int maxLength) {
-    String cleansed = val.replaceAll("[^A-Za-z0-9\\-_]", "-").toLowerCase();
+    String cleansed = val.replaceAll("[^A-Za-z0-9\\-]", "-").toLowerCase();
     return cleansed.length() > maxLength ? cleansed.substring(0, maxLength) : cleansed;
   }
 
   /**
-   * Deploys a {@link V1Deployment} to for runnable exeuction in Kubernetes.
+   * Deploys a {@link V1Job} to for runnable execution in Kubernetes.
+   */
+  private V1ObjectMeta createJob(V1ObjectMeta metadata, Map<String, RuntimeSpecification> runtimeSpecs,
+                                 Location runtimeConfigLocation) throws ApiException {
+    BatchV1Api batchApi = new BatchV1Api(apiClient);
+    V1Job job = new V1JobBuilder()
+      .withMetadata(metadata)
+      .withNewSpec()
+      .withManualSelector(true)
+      .withSelector(new V1LabelSelector().matchLabels(metadata.getLabels()))
+      // TODO:CDAP-18125 Change parallelism based on worker instances
+      .withParallelism(1)
+      .withCompletions(1)
+      .withBackoffLimit(0)
+      .withNewTemplate()
```

Comment:
nit: it's nice to follow the indentation we have elsewhere, where everything for the template is indented more until endTemplate()

same for withNewSpec()